core/templates/dev/head/domain/objects/NumberWithUnitsObjectFactory.js

```diff
@@ -129,17 +146,17 @@ oppia.factory('NumberWithUnitsObjectFactory', [
       return new NumberWithUnits(
         numberWithUnitsDict.type,
         numberWithUnitsDict.real,
-        numberWithUnitsDict.fraction,
-        numberWithUnitsDict.units);
+        FractionObjectFactory.fromDict(numberWithUnitsDict.fraction),
+        UnitsObjectFactory.fromList(numberWithUnitsDict).units);
```

Comment:
You initialization is inconsistent. You are passing unitsObj [above](https://github.com/oppia/oppia/pull/5028/files#diff-6c6f349899682c0c4431532a6728d573R142), whereas you are passing unitsObj.units here.

Reply:
No, while creating object above it's calling `NumberWithUnits` constructor which is assigning unitsObj.units value. This has been confirmed with unit tests as well.

Reply:
It is calling NumberWithUnits constructor at both places. The method above on line 142 is passing unitsObj object while the line at which this comment is placed is passing unitsObj.units. Isn't that inconsistent?

Also, in the constructor function, you are assigning `units = unitsObj.units`. Which means that it expects unitsObj and not a list (list is what you are passing here).

Reply:
Oh yes. I haven't realised it before. I'll modify it.

Reply:
Done.